### PR TITLE
podvm: disable cloud-init unsafe modules for CoCo

### PR DESCRIPTION
**Pull request by snir911**
this is executed when CONFIDENTIAL_COMPUTE_ENABLED=yes or based on the EXPLICIT_adapt_cloud_init value, if set

in either case, this is currently disabled for the libvirt provider

Fixes: [KATA-3269](https://issues.redhat.com//browse/KATA-3269)

### CODE REVIEW

File: config/peerpods/podvm/lib.sh [hunk 1]
```diff
@@ -229,6 +229,11 @@ function prepare_source_code() {
         create_overlay_mount_unit
```
Comment (bpradipt) [inline config/peerpods/podvm/lib.sh:229]: I think `adapt_cloud_init` can be called here ..

**Review by gkurz — COMMENTED** [review]
@snir911 what's the `tmporary change` commit for ?

**Comment by snir911** [comment]
tmporary change was to test wheter this is works with libvirt provider, currently thanks to @ajayvic it seems to be failing so i removed this patch for now
```
[  OK  ] Reached target Cloud-init target.
         Starting Agent Protocol Forwarder...
         Starting Record Runlevel Change in UTMP...
[  OK  ] Finished Record Runlevel Change in UTMP.
[FAILED] Failed to start Agent Protocol Forwarder.
See 'systemctl status agent-protocol-forwarder.service' for details.
[  OK  ] Started Confidential Data Hub TTRPC API Server.
```

File: config/peerpods/podvm/lib.sh (hunk 1 above)
Comment (snir911) [reply config/peerpods/podvm/lib.sh:229]: not an option ATM as this should be disabled for libvirt

Comment (bpradipt) [reply config/peerpods/podvm/lib.sh:229]: How about something like the following ?
[[ "$CLOUD_PROVIDER" != "libvirt" ]] && adapt_cloud_init

Comment (bpradipt) [reply config/peerpods/podvm/lib.sh:229]: If you want the ability to enable/disable the cloud-init modules explicitly then it makes sense to have it separately..
Can you please change the config param to CUSTOM_CLOUD_INIT_MODULES and the function name to set_custom_cloud_init_modules

File: config/peerpods/podvm/lib.sh [hunk 2]
```diff
@@ -229,6 +229,11 @@ function prepare_source_code() {
...
     fi
 
+    # disable ssh and unsafe cloud-init modules
+    if [[ "$CONFIDENTIAL_COMPUTE_ENABLED" == "yes" ]] || [[ -n "$CUSTOM_CLOUD_INIT_MODULES" ]]; then
+       [[ "$CUSTOM_CLOUD_INIT_MODULES" != "no" ]] && [[ "$CLOUD_PROVIDER" != "libvirt" ]] && set_custom_cloud_init_modules
```
Comment (bpradipt) [inline config/peerpods/podvm/lib.sh:234]: where is `CUSTOM_CLOUD_INIT_MODULES` set ?

Comment (bpradipt) [reply config/peerpods/podvm/lib.sh:234]: Ok, so it's there only for disabling..

**Review by bpradipt — APPROVED** [review]
/lgtm

**Review by gkurz — APPROVED** [review]
Under the assumption that the content of `99_coco_only_allow.cfg` is correct, the patch looks good to me.

Thanks @snir911 !

**Pull request merged**